mne/decoding/tests/test_time_gen.py

```diff
@@ -70,7 +70,15 @@ def test_generalization_across_time():
     assert_equal("<GAT | fitted, start : -0.200 (s), stop : 0.499 (s), no "
                  "prediction, no score>", '%s' % gat)
     assert_equal(gat.ch_names, epochs.ch_names)
+    # test different predict function:
+    gat.decision_function(epochs)
+    assert_array_equal(np.shape(gat.y_pred_), (15, 15, 14, 1))
+    gat.predict_proba(epochs)
+    assert_array_equal(np.shape(gat.y_pred_), (15, 15, 14, 2))
```

Comment:
I guess that the only option left.. Is this what they do in sklearn? That still seems quite complicated.

I must say I don't quite get the overall logic. `predict`, `predict_proba` and `decision_function` are just different prediction formats (discrete, continuous in original metrics, continuous in probabilistic metric) aren't they. IMO they should all output in the same variable `y_pred`, but I guess it's not the place to discuss sklearn API.